TestFoundation/TestNSCalendar.swift

```diff
@@ -0,0 +1,906 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+enum TestError: Error {
+    case unexpectedNil
+}
+
+extension Optional {
+    func unwrapped() throws -> Wrapped {
+        if let x = self {
+            return x
+        } else {
+            XCTFail()
+            throw TestError.unexpectedNil
+        }
+    }
+}
```

Comment:
Probably should move these into `TestFoundation/Utilities.swift` as they could be reused elsewhere and are not specific to testing calendars.

Also will the `XCTFail()` here will not give a very useful error message as it won't show the location of the caller. Maybe something like this would be better:
```
    func unwrapped(_ fn: String = #function, file: StaticString = #file, line: UInt = #line) throws -> Wrapped {
       if let x = self {
            return x
        } else {
            XCTFail("unwrapped() failed in \(file):\(line):\(fn)")
            throw TestError.unexpectedNil
        }
    }
```

Reply:
Yeah, that makes more sense.

In general I want to move away from force unwraps in tests because crashing the TestFundatin process can stop executing subsequent tests, and this is an attempt to do better.